src/tensors/tensor.jl

```diff
@@ -389,7 +389,7 @@ function Base.complex(t::AbstractTensorMap)
     if scalartype(t) <: Complex
         return t
     else
-        return copy!(similar(t, complex(scalartype(t))), t)
+        return TensorMap(complex(t.data), space(t))
```

Comment:
I'm not a big fan of having `TensorMap` hard-coded here. I think `similar`, by definition, should return a mutable tensor to copy the data into, which by default returns a `TensorMap`, similar to how `Array` is the default for `similar(::AbstractArray)`, but I need to overload this for `BlockTensorMap`, which is easier if everything passes through similar.

Reply:
It feels like it's the second time I make this mistake. I forget that this method works on `AbstractTensorMap`, whereas it is in tensormap.jl. Maybe I should move it. However, I do think having the simple `TensorMap` specialisation around is also useful.